File: simple_split_frontend/lib/models/expense.dart
class Expense {
  final String id;
  final String groupId;
  final String payerId; // Mudado de int para String (UUID)
  final String? payerName;
  final String description;
  final double amount;
  final DateTime? date;
  final DateTime? createdAt;
  final List<Debt>? debts;

  Expense({
    required this.id,
    required this.groupId,
    required this.payerId,
    this.payerName,
    required this.description,
    required this.amount,
    this.date,
    this.createdAt,
    this.debts,
  });

  factory Expense.fromJson(Map<String, dynamic> json) {
    return Expense(
      id: json['id'].toString(),
      groupId: json['group_id'].toString(),
      payerId: json['payer_id'].toString(), // Garantir que seja String
      payerName: json['payer_name'],
      description: json['description'],
      amount: json['amount'].toDouble(),
      date: json['date'] != null ? DateTime.parse(json['date']) : null,
      createdAt: json['created_at'] != null 
          ? DateTime.parse(json['created_at']) 
          : null,
      debts: json['debts'] != null 
          ? (json['debts'] as List).map((d) => Debt.fromJson(d)).toList()
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'group_id': groupId,
      'payer_id': payerId,
      'payer_name': payerName,
      'description': description,
      'amount': amount,
      'date': date?.toIso8601String().split('T')[0],
      'created_at': createdAt?.toIso8601String(),
      'debts': debts?.map((d) => d.toJson()).toList(),
    };
  }
}

class Debt {
  final String id;
  final String expenseId;
  final String debtorId; // Mudado de int para String (UUID)
  final String? debtorName;
  final String creditorId; // Mudado de int para String (UUID)
  final String? creditorName;
  final double amount;
  final String status;
  final DateTime? dueDate;
  final DateTime? paidAt;
  final DateTime? createdAt;
  final String? expenseDescription;

  Debt({
    required this.id,
    required this.expenseId,
    required this.debtorId,
    this.debtorName,
    required this.creditorId,
    this.creditorName,
    required this.amount,
    required this.status,
    this.dueDate,
    this.paidAt,
    this.createdAt,
    this.expenseDescription,
  });

  factory Debt.fromJson(Map<String, dynamic> json) {
    return Debt(
      id: json['id'].toString(),
      expenseId: json['expense_id'].toString(),
      debtorId: json['debtor_id'].toString(), // Garantir que seja String
      debtorName: json['debtor_name'],
      creditorId: json['creditor_id'].toString(), // Garantir que seja String
      creditorName: json['creditor_name'],
      amount: json['amount'].toDouble(),
      status: json['status'],
      dueDate: json['due_date'] != null 
          ? DateTime.parse(json['due_date']) 
          : null,
      paidAt: json['paid_at'] != null 
          ? DateTime.parse(json['paid_at']) 
          : null,
      createdAt: json['created_at'] != null 
          ? DateTime.parse(json['created_at']) 
          : null,
      expenseDescription: json['expense_description'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'expense_id': expenseId,
      'debtor_id': debtorId,
      'debtor_name': debtorName,
      'creditor_id': creditorId,
      'creditor_name': creditorName,
      'amount': amount,
      'status': status,
      'due_date': dueDate?.toIso8601String().split('T')[0],
      'paid_at': paidAt?.toIso8601String(),
      'created_at': createdAt?.toIso8601String(),
      'expense_description': expenseDescription,
    };
  }

  bool get isPending => status == 'pending';
  bool get isPaid => status == 'paid';
  bool get isCancelled => status == 'cancelled';
}

File: simple_split_backend/app/models/expense.py
from app import db
from datetime import datetime
import uuid

class Expense(db.Model):
    __tablename__ = 'expenses'
    
    id = db.Column(db.String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    group_id = db.Column(db.String(36), db.ForeignKey('groups.id'), nullable=False)
    payer_id = db.Column(db.String(36), db.ForeignKey('users.id'), nullable=False)
    description = db.Column(db.String(200), nullable=False)
    amount = db.Column(db.Float, nullable=False)
    date = db.Column(db.Date, default=datetime.utcnow().date())
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    
    # Relacionamentos
    debts = db.relationship('Debt', backref='expense', cascade='all, delete-orphan')
    
    def to_dict(self):
        return {
            'id': self.id,
            'group_id': self.group_id,
            'payer_id': self.payer_id,
            'payer_name': self.payer.name if self.payer else None,
            'description': self.description,
            'amount': self.amount,
            'date': self.date.isoformat() if self.date else None,
            'created_at': self.created_at.isoformat() if self.created_at else None,
            'debts': [debt.to_dict() for debt in self.debts]
        }
    
    def split_expense(self, member_ids=None):
        """Divide a despesa entre os membros do grupo"""
        # Se não especificou membros, divide entre todos do grupo
        if not member_ids:
            from app.models.user import GroupMember
            group_members = GroupMember.query.filter_by(group_id=self.group_id).all()
            member_ids = [member.user_id for member in group_members]
        
        # Remove o pagador da lista se ele estiver incluído
        if self.payer_id in member_ids:
            member_ids.remove(self.payer_id)
        
        if not member_ids:
            return  # Ninguém deve nada
        
        # Calcula o valor que cada um deve
        amount_per_person = self.amount / (len(member_ids) + 1)  # +1 para incluir o pagador
        
        # Criar dívidas
        from app.models.debt import Debt
        for debtor_id in member_ids:
            debt = Debt(
                expense_id=self.id,
                debtor_id=debtor_id,
                creditor_id=self.payer_id,
                amount=amount_per_person,
                status='pending'
            )
            db.session.add(debt)
        
        db.session.commit()